src/calendar/calendar.tsx

```diff
@@ -99,56 +101,108 @@ export const Calendar = ({
     };
 
     const toggleMonthView = () => {
-        setCurrentView("month");
+        if (currentView !== "month-options") {
+            setCurrentView("month-options");
+        } else {
+            setCurrentView("default");
+        }
     };
 
     const toggleYearView = () => {
-        setCurrentView((prev) => {
-            if (prev !== "day") return "day";
-
-            return "year";
-        });
+        if (currentView !== "year-options") {
+            setCurrentView("year-options");
+        } else {
+            setCurrentView("default");
+        }
     };
 
-    // =============================================================================
-    // RENDER FUNCTIONS
-    // =============================================================================
-    const renderYearHeader = () => {
-        let displayText: string;
-        if (currentView === "year") {
+    const getYearHeaderText = (): string => {
+        if (currentView === "year-options") {
             const { beginDecade, endDecade } =
                 CalendarHelper.getStartEndDecade(calendarDate);
 
-            displayText = `${beginDecade} to ${endDecade}`;
+            return `${beginDecade} to ${endDecade}`;
         } else {
-            displayText = dayjs(calendarDate).format("YYYY");
+            return dayjs(calendarDate).format("YYYY");
         }
+    };
 
-        return <Text.H4 weight="regular">{displayText}</Text.H4>;
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const renderCalendarContent = () => {
+        switch (currentView) {
+            case "month-options":
+                return null;
+            case "year-options":
+                return null;
+            default:
+                return (
+                    <CalendarDay
+                        calendarDate={calendarDate}
+                        disabledDates={disabledDates}
+                        selectedStartDate={selectedStartDate}
+                        onSelect={handleDateSelect}
+                    />
+                );
+        }
     };
 
     return (
         <Container $type={type} {...otherProps}>
-            <HeaderDropdown $view={currentView}>
+            <SideArrowButton $direction="left" onClick={handleLeftArrowClick}>
```

Comment:
Instead of adding false margin, we should position them correctly to make full use of `justify-content` style in `Container`. I also added a larger hit zone for the side arrow buttons for easier clicking